pkg/kv/kvserver/rangefeed/processor_test.go

```diff
@@ -634,6 +635,45 @@ func TestProcessorMemoryBudgetReleased(t *testing.T) {
 	})
 }
 
+// TestIntentScannerOnError tests that when a processor is given with an intent
+// scanner constructor that fails to create a scanner, the processor will fail
+// to start gracefully. Regression for #127204.
+func TestIntentScannerOnError(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	t.Helper()
+	stopper := stop.NewStopper()
+	cfg := testConfig{
+		Config: Config{
+			RangeID: 2,
+			Stopper: stopper,
+			Span:    roachpb.RSpan{Key: roachpb.RKey("a"), EndKey: roachpb.RKey("z")},
+			Metrics: NewMetrics(),
+		},
+	}
+	sch := NewScheduler(SchedulerConfig{
+		Workers:         1,
+		PriorityWorkers: 1,
+		Metrics:         NewSchedulerMetrics(time.Second),
+	})
+	require.NoError(t, sch.Start(context.Background(), stopper))
+	cfg.Scheduler = sch
+	s := NewProcessor(cfg.Config)
+	require.ErrorContains(t, s.Start(stopper, func() (IntentScanner, error) {
+		return nil, errors.New("scanner error")
+	}), "scanner error")
+	ctx := context.Background()
+	defer stopper.Stop(ctx)
+	testutils.SucceedsSoon(t, func() error {
+		select {
+		case <-s.(*ScheduledProcessor).stoppedC:
+			return nil
```

Comment:
Are there any other invariants we can check here?

Reply:
I added checks for unregistering the processor from the scheduler and for rts.IsInit(). However, I realized that this isn’t testing what I want to test here since the resolved timestamp will only be initialized much later (after we forward closed timestamp). I tried to add a new case where the intent scanner operates correctly and the resolved timestamp initializes successfully. However, I found the implementation to be more complex than expected, effectively requiring us to reinvent the logic in [TestProcessorInitializeResolvedTimestamp](https://github.com/cockroachdb/cockroach/blob/d5f6b23bad0c1c798c49b02db7bb89ea8abe2d1a/pkg/kv/kvserver/rangefeed/processor_test.go#L685). If we feel strong about this, I can refactor the TestProcessorInitializeResolvedTimestamp test to exercise both error and non-error cases.

Reply:
Removed the assertion on rts.IsInit and left unregistering processor from scheduler in the test. Not sure how much you would love it - I'm checking by accessing the scheduler maps here. 

```
_, ok := sch.shards[shardIndex(p.ID(), len(sch.shards), p.Priority)].procs[p.ID()]
require.False(t, ok)
require.False(t, sch.priorityIDs.Contains(p.ID()))
```